pkg/logs/internal/tailers/channel/tailer.go

```diff
@@ -83,11 +89,15 @@ func (t *Tailer) run() {
 }
 
 func computeServiceName(lambdaConfig *config.Lambda, serviceName string) string {
-	if lambdaConfig == nil {
-		return "agent"
-	}
-	if len(serviceName) > 0 {
-		return strings.ToLower(serviceName)
+	if isServerlessOrigin(lambdaConfig) {
+		if len(serviceName) > 0 {
+			return strings.ToLower(serviceName)
+		}
+		return strings.ToLower(os.Getenv(cloudRunServiceName))
 	}
-	return ""
+	return "agent"
```

Comment:
Just to double-check, since this happens outside of serverless block is this change to the default service name intentional?